stock_picking_update_responsible/models/stock_picking.py

```diff
@@ -0,0 +1,18 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+
+from odoo import models, fields
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    is_user_id_editable = fields.Boolean(
+        compute="_compute_is_user_id_editable",
+    )
+
+    def _compute_is_user_id_editable(self):
+        is_user_id_editable = self.env.user.has_group(
+            "stock_picking_update_responsible.group_stock_picking_preparer"
+        ) or self.env.user.has_group("sales_team.group_sale_salesman")
+        self.is_user_id_editable = is_user_id_editable
```

Comment:
Remember that readonly in view has already a condition (depending on picking state), if we don't add it here, it will be lost and we'll changing default behavior.

And also, group belonging condition should be added only for sales related pickings (e.g. if picking comes from a purchase or an internal operation, current behavior shouldn't be changed)